### PR TITLE
SIG SG magazines and rifles

### PR DESCRIPTION
**Pull request by SnowFreeTrack**
#### Summary
SUMMARY: Content "Adds SIG SG magazines and rifle(s)"

#### Purpose of change
A new magazine type as well as a new rifle to support it. Changed the name of the 552 to 553 as both of the variants are about the same (553 being an upgrade to 552). 

#### Additional context
Not sure if the makeshift SIG SG magazine is needed with the item probabilities (not for me to decide but anyway) being so low.

### CODE REVIEW

File: data/json/itemgroups/guns.json [hunk 1]
```diff
@@ -572,4 +575,4 @@
       { "item": "sks", "prob": 10, "charges-min": 0, "charges-max": 10 }
     ]
   }
-]
+]
```
Comment (anothersimulacrum) [inline data/json/itemgroups/guns.json]:
```suggestion
]

```

File: data/json/items/gun/223.json [hunk 2]
```diff
@@ -518,4 +541,4 @@
     ],
     "magazines": [ [ "223", [ "survivor223mag", "stanag30", "stanag50" ] ] ]
   }
-]
+]
```
Comment (anothersimulacrum) [inline data/json/items/gun/223.json]:
```suggestion
]

```

File: data/json/items/magazine/223.json [hunk 3]
```diff
@@ -177,4 +209,4 @@
     "name": "Ruger makeshift magazine",
     "description": "An improvised 5-round straight single-stack box magazine consisting of little more than a bent sheet of steel held together by duct tape and hope, for use with the Ruger Mini-14 rifle."
   }
-]
+]
```
Comment (anothersimulacrum) [inline data/json/items/magazine/223.json]:
```suggestion
]

```

**Comment by tenmillimaster** [comment]
Please add entries for the new gun and magazines to generic guns' migrations. (/data/mods/generic guns)
I don't believe there's any need to change the ID of sig552 to sig553, but if really needed, you'll also need to add a migration entry in the main game.

Volume analysis:
For the sg550:
 Length of rifle: 998mm Width of rifle used (37.3564mm) 
Width: 56mm([width of this B+T handguard](https://hkparts.net/product/b-t-sig-552-553-handguard-p16870.htm))- ([minimum height of two picatinny rails](https://upload.wikimedia.org/wikipedia/commons/thumb/6/6a/M1913A_Rail_CrossSection.svg/800px-M1913A_Rail_CrossSection.svg.png)) 
image

Volume: 74812.984 square mm * 37.3564 mm = ~2795 mL.


For the sg553:
 Length of rifle: 730mm Width of rifle used (37.3564mm, see above) 
image

Volume: 57762.957 square mm * 37.3564mm = ~ 2158 mL



The SG550 doesn't need a damage penalty(so it's ranged damage should be 0), and the 553 should get a -6. 
image

None of the SG55x rifles belong in the military item groups.

**Comment by SnowFreeTrack** [comment]
@tenmillimaster Great way to measure volume! I'll start using it from now on, perhaps even fix some other firearms while I'm at it. Also, that's a handy spreadsheet for damage calculation. Is it available somewhere? Or is it your patented formula?

**Comment by tenmillimaster** [comment]
Any ball ammunition is defined as doing the square root of its muzzle energy. I just adapted 
# work from Dr. Phillip Dater of Gemtech Suppressors[work from Dr. Phillip Dater of Gemtech Suppressors](https://ndiastorage.blob.core.usgovcloudapi.net/ndia/2010/armament/WednesdayCumberlandPhilipDater.pdf). I just select whichever regression works best for the graph, forcing an y intercept of 0 (r value as close as possible to 1) and call it good enough.

The program to get the 'area' of a gun's profile is imagej, you just need a good picture of it and a known length. Width is usually trickier.


Note, you still need to update the migration in /data/json/items? to swap out all of the 552's for 553's. You should also consider removing some of the slots on the sigs, since they don't come standard with siderails or optic rails.

**Comment by SnowFreeTrack** [comment]
@tenmillimaster You read my mind! I was just thinking about the slots. I replaced the 552 in Generic gun mod and either I'm blind (which might as well be the case) or there are no other 552's left. Also, while on the subject of volume, it is actually possible to make an approximate calculation of the area using Photoshop or Gimp, which could be useful for future volume correction or calculation.

**Comment by tenmillimaster** [comment]
People's old saves will still reference `sig552`, hence the migration to replace them with the new ID.

**Comment by ZhilkinSerg** [comment]
Feel free to resolve conflicts and reopen if you feel working on it again.

**Pull request closed without merging**